dstep/translator/Type.d

```diff
@@ -350,9 +352,10 @@ body
         {
             if (children[dimension].kind == CXCursorKind.CXCursor_IntegerLiteral)
             {
+                auto token = tokenInsideSquareBrackets(cursor);
                 auto expansions = context.macroIndex.queryExpansion(children[dimension]);
 
-                if (expansions.length == 1)
+                if (expansions.length == 1 && token.kind == TokenKind.identifier)
```

Comment:
The Clang AST for this is... weird. I'm slightly surprised it gets translated to the right machine code even. The `FieldDecl` ends up having an `IntegerLiteral` child with empty spelling, which on the face of it makes no sense (to me). And yet, this is what libclang gives us.

I'll figure out how to deal with your extended example. It looks like it'll be complicated.